JCSDA/spack-stack
# Add weekly build/test scripts

### Summary
This PR will add scripting for weekly unit testing and build caching of the unified env through Jenkins, cron jobs, etc.

Also a few updates for Acorn, including a change to `spack stack create env --compiler` that allows compiler version to be specified (`spack stack create env --compiler intel` -> `envs/myenv-intel`; `spack stack create env --compiler intel@2022` -> `envs/myenv-intel-2022`), which is needed to differentiate on ~~platforms~~ platform where more than one version of a compiler is installed.

### Testing/implementation by platform
On each system:
- Configure ShellSetup.sh (including setting centralized build_cache dir under spack-stack/ dir)
- Set up centralized source cache
- Get it running (Jenkins, cron, GHA); test installing a few packages
- Enable notifications (email, Slack, whatever)

Platforms (check when all of the above items are complete):
 - [x] Acorn - @AlexanderRichert-NOAA - cron
 - [x] Orion - @rickgrubin-tomorrow test with Jenkins on Orion
 - [x] Nautilus - @climbfuji - either cron or Github actions

### Applications affected
all

### Systems affected
all (eventually)

### Dependencies
none

### Issue(s) addressed
#1184, #365

### Checklist
- [x] This PR addresses one issue/problem/enhancement, or has a very good reason for not doing so.
- [x] These changes have been tested on the affected systems and applications.
- [ ] All dependency PRs/issues have been resolved and this PR can be merged.
